Update gradle

Aracem opened the pull request:
### :tophat: What is the goal?

Update Gradle (tools, wrapper..) to improve the compilation time and take advantage of the last improvements in Android Studio 3.0.
Also, update dependencies to follow new implementation/api instead of compile standard, that helps to improve compilation times

### How is it being implemented?

* Follow this tutorial 
https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration?utm_source=android-studio#new_configurations
* Add google() repository 
* Set compile version and target version to 27 
* Create variables in gradle to update the versions of the library and the samples at the same time

### GIF

![happy](https://media.giphy.com/media/YIW0KqAQShjCE/giphy.gif)

Aracem committed:
Upgrade target and compile SDK versions to 27

Also, add a common configuration with variables in the Rosie gradle file
to easily change this configurations in the library and the samples at the
same time.

Aracem committed:
Upgrade gradle wrapper to v4.4 and gradle tools to v3.1.3

Also, added google() repository to gradle because now it's mandatory.

Aracem committed:
Update gradle files according new Android Gradle plugin 3.0

Please, review this tutorial for a better understanding of this commit:
  https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration

Apply this new rules to the gradle files, changing compile by implementation
and adding annotationProcessor. Also, edit the order of the repositories
jcenter() and google() to avoid download dependencies errors.

Aracem commented on rosie/build.gradle:
```diff
@@ -18,26 +18,27 @@ apply plugin: 'com.android.library'
 
-  provided 'com.squareup.dagger:dagger-compiler:1.2.2'
+  annotationProcessor 'com.squareup.dagger:dagger-compiler:1.2.2'
+  compileOnly 'com.squareup.dagger:dagger-compiler:1.2.2'
```
I don't know why the dagger dependency is added twice but I maintain them.

pedrovgs replied:
I don't remember why xD

pedrovgs reviewed (changes requested):
Thank you so much @Aracem. This PR is really helpful. I was wondering if you could update the Gradle Wrapper this project contains. As we've updated the Gradle version, the wrapper should be updated too 😃

Aracem commented:
Should be updated:
https://github.com/Karumi/Rosie/pull/97/files#diff-fd9d61e37d939c941e7a62077f38d62dR22

I have to add a modification to make the CI accept the new Terms And Conditions when updating it. Let me ask a friend of mine about it

pedrovgs commented:
You are absolutely right @Aracem the wrapper is updated. I didn't see the file, my bad. We will wait until the build passes to merge this branch. Thank you so much 😃

Aracem commented:
The builds are broken due the com.android.library plugin, that download dependencies but Google adds a "accepts terms and conditions" question while downloading, so you have to configure CI to accept it.

Here is the solution: https://stackoverflow.com/a/45782695/628447
```yes | sdkmanager --licenses```

Aracem committed:
Fix checkstyle errors

Aracem committed:
Add an option in gradle to fix a downloading problem with Roboelectric

Comming from here:
https://github.com/robolectric/robolectric/issues/3452#issuecomment-386794265

Aracem commented:
Please @pedrovgs review [commit](https://github.com/Karumi/Rosie/pull/97/commits/4eae6d463d3e60ee7d3846adc4bb7b17cb3ff8e4)  and [commit](https://github.com/Karumi/Rosie/pull/97/commits/766b6b2b9650ec011f495e539bfe0990ffaa4f50) before approving the PR. I had to introduce these changes due Checkstyle errors

I need help to resolve the Robolectric error. Let's see how we can resolve it
